typescript/sdk/src/token/Token.ts

```diff
@@ -211,6 +211,10 @@ export class Token implements IToken {
       return new EvmHypCollateralAdapter(chainName, multiProvider, {
         token: addressOrDenom,
       });
+    } else if (standard === TokenStandard.EvmHypCollateralFiat) {
+      return new EvmHypCollateralAdapter(chainName, multiProvider, {
```

Comment:
If this contract is not **actually** compatible with `EvmHypCollateralAdapter` (as implied by the need for your other PR), then I think a better solution would be to make a new TokenAdapter for it. It can extend `EvmHypCollateralAdapter` with whatever special handling you need for collateral balance checking

Reply:
Do I still need `EvmHypCollateralAdapter`, it won't have any logic because the frontend uses the `isDestinationCollateralSufficient()`?

Reply:
What solution are you suggesting instead? Are the current changes here sufficient to solve the problem without UI-level changes?

Reply:
Yup, no UI changes are needed. 

CollateralFiat is a bit strange. It acts like a Collateral Token (expects an underlying collateral), and synthetic (no balance on the Token itself as it burns the collateral right away).

Reply:
Heres what i mean for `isDestinationCollateralSufficient()`

https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/0d544d97e5fe77feaa98b9147fa38b6da4516db5/typescript/sdk/src/warp/WarpCore.ts#L472-L481